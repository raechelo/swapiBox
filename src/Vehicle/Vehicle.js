import React, { Component } from 'react';
import propTypes from 'prop-types';
import { fetchCalls } from '../apiCalls';


class Vehicle extends Component {
  constructor() {
    super()
    this.state = {
      vehicles: []
    }
  }

  componentDidMount() {
    const vehicleUrl = 'https://swapi.co/api/vehicles/';
    return fetchCalls(vehicleUrl)
      .then(data => this.setState( {vehicles: data.results} ) )
      .catch(err => { throw new Error(err) } )
  }
  

  render() {
    const displayVehicles = this.state.vehicles.map(vehicle => (
      <article className="Card">
        <h4>{vehicle.name}</h4>
        <h6>Model: {vehicle.model}</h6>
        <h6>Class: {vehicle.class}</h6>
        <h6>Passenger Count: {vehicle.passengers}</h6>
        <h6><i onClick={() => this.props.favoriteItem( vehicle ) } class="far fa-star"></i></h6>
      </article>
    ))

    return (
      <section className="Card-Container">
        {displayVehicles}
      </section>
    )
  }
}

Vehicle.propTypes = {
  v: propTypes.object,
  favoriteItem: propTypes.func,
  name: propTypes.string,
  model: propTypes.string,
  passengers: propTypes.number,
  class: propTypes.string
}

export default Vehicle;