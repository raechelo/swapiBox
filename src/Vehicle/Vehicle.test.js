import React from 'react';
import { shallow } from 'enzyme';
import Vehicle from './Vehicle';

describe('Vehicle', () => {

  let wrapper;

  let mockName = 'Nimbus';
  let mockModel = '2000';
  let mockPassengers = 1;
  let mockVehicle_class = 'flying broom'

  beforeEach(() => {
    wrapper = shallow (
      <Vehicle name={mockName} model={mockModel} passengers={mockPassengers} class={mockVehicle_class} />
    )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })
})