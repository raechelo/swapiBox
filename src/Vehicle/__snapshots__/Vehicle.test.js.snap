// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicle should match the snapshot with all the data passed in 1`] = `
<div
  className="Card"
>
  <h4>
    Nimbus
  </h4>
  <h6>
    Model: 
    2000
  </h6>
  <h6>
    Class: 
    flying broom
  </h6>
  <h6>
    Passenger Count: 
    1
  </h6>
  <h6>
    <i
      class="far fa-star"
    />
  </h6>
</div>
`;
